### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by cyril-sf**

### CODE REVIEW

**Comment by cyril-sf** [comment]
@hjdivad good for review

File: README.md [hunk 1]
```diff
@@ -1,14 +1,43 @@
...
 
 It is important to note that **the MessageChannel.js communication can only work
 with browser that supports `window.postMessage`.**
 
+## Usage
+
+The library works almost like a polyfill.
+
+The only difference
+in the code is the way `MessagePort` are sent between user agents.
```
Comment (hjdivad) [inline README.md]: `MessagePort`s.

File: README.md [hunk 2]
```diff
@@ -1,14 +1,43 @@
...
+
+```js
+ Window.postMessage(otherWindow, 'hello', "http://example.com", [channel.port1]);
+ Worker.postMessage(worker, 'hello', "http://example.com", [channel.port1]);
+```
+
+## Limitations
+
+- A `MessagePort` can be transfered to another user agent that isn't a parent or a child.
+In that specific use case, every intermediate user agent needs to keep existing and can access
```
Comment (hjdivad) [inline README.md]: I think you accidentally a noun, or maybe some other Frenglish.

How about

> In that case, each intermediate user agent needs to a) persist and b) retain a handle to the user agent it transferred the port to.  So if port `p` is sent from agent `a` to `c` via `b`, `b` will still see messages between `a` and `c`.

**Comment by hjdivad** [comment]
@Cyril-sf seems good, sans some minor Frenglish.

File: README.md (hunk 2 above)
Comment (cyril-sf) [inline README.md]: You forgot a word too, no ("I think you accidentally a noun") ? :smile: 

Anyway, that seems clearer.

Comment (hjdivad) [inline README.md]: Are you referring to "accidentally a noun"?  If so, no, [it's a meme](http://knowyourmeme.com/memes/i-accidentally)

Comment (cyril-sf) [inline README.md]: :thumbsup:

**Pull request merged**